<launch>
    <!-- robot model -->
    <include file="$(find codbot_description)/launch/codbot_description.launch"/>
	<arg name="base_type"       default="$(env BASE_TYPE)" />
    <node name="codbot_base_node" pkg="codbot_base" type="codbot_base_node" output="screen">
        <!--debug num: 8-bit binary number ,1;0000 0001=1 2:0000 0010=2 3:0000 0100=4 4:0000 1000=8-->
        <param name="debug_num"	value="3"/><!--You can also combine numbers-->
        
		<!-- serial_com set-->
        <param name="baud_rate"   	value="B115200"/>
        <param name="cmd_port" 	value="/dev/base"/>

		<!--odom coefficient of xy-->
		<param name="coeff_odom_x" 		value="1.061"/>
		<param name="coeff_odom_y" 		value="1.077"/>
		
        <!--absolute value of velocity limit -->
        <param name="cmd_vel_linear_max"   	value="1.0"/><!-- unit:m/s -->
		<param name="cmd_vel_linear_min"   	value="0.1"/><!-- unit:m/s -->
        <param name="cmd_vel_anglar_max"  	value="2"/><!-- unit:rad/s -->
		<param name="cmd_vel_anglar_min"  	value="0.00"/><!-- unit:rad/s -->
		<param name="cmd_vel_angle_max"	value="25"/><!--unit: rad (only ackerman)-->
		<param name="cmd_vel_angle_min"	value="1"/><!--unit: rad (only ackerman)-->		

		<!--current_type A100 D100 T100 M100-->
        <param name="current_type"     	 	value="$(arg base_type)"/>

		<!--topic and frame-->
        <param name="cmd_vel_topic"                	value="/cmd_vel"/>
        <param name="odom_topic"               		value="/odom"/>
        <param name="imu_topic"                		value="/imu"/>
        <param name="wheel_left_speed_pub_topic"   	value="wheel_left_speed"/>
        <param name="wheel_right_speed_pub_topic"  	value="wheel_right_speed"/>
        <param name="imu_frame"                 	value="imu_link"/>
        <param name="odom_frame"                	value="odom"/>
        <param name="odom_child_frame"          	value="base_footprint"/>
    </node>


    <!--node pkg = "robot_pose_ekf" type = "robot_pose_ekf" name = "robot_pose_ekf" output = "screen">
        <param name = "output_frame" value="odom" />
        <param name = "base_footprint_frame" value="base_footprint" />
        <param name = "freq" value="40.0" />
        <param name = "debug" value="false" />
        <param name = "sensor_timeout" value ="1.0" />
        <param name = "odom_used" value ="true" />
        <param name = "imu_used" value ="true" />
        <param name = "vo_used" value ="false" />
        <param name = "gps_used" value ="false" />
        <remap from = "/odom" to ="/odom_base" />
        <remap from = "/imu_data" to = "/imu" />
    </node>

    <node pkg = "robot_pose_ekf" type = "odom_ekf.py" name = "odom_ekf" output = "screen">
        <remap from = "input"  to = "/robot_pose_ekf/odom_combined" />
        <remap from = "output" to = "/odom" />
    </node-->
</launch>
